src/background_thread.c

```diff
@@ -609,88 +573,23 @@ background_threads_disable(tsd_t *tsd) {
 	return false;
 }
 
-/* Check if we need to signal the background thread early. */
+bool
+background_thread_running(background_thread_info_t *info) {
```

Comment:
Agree, using `running` is incorrect with current implementation.
It's a bit confusing that thread state is set to `started` even before it's created, but I'm not sure it can be simplified without bloating the code with unnecessary states.

Reply:
I wish I could do `background_thread_started`, but it's already an existing symbol from `enum background_thread_state_t`. What about `background_thread_is_started`? Or I can remove that whatsoever and revert to `==`, but my gut tells me that thread state should be internal to `background_thread`.

Reply:
`_is_started` sounds good to me.

re: why it's set to `started` before it's created -- the context is we want to avoid holding any locks when calling pthread_create, while we also want make sure only one thread is created for the specific index. Therefore marking it as started while holding the index lock. Like you said it can use another "creating" state, but the value is a bit limited given that it's a one-time transition (and usually very quick).